fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/TypeRdfContext.java

```diff
@@ -58,71 +61,36 @@ public TypeRdfContext(final FedoraResource resource,
 
         //include rdf:type for primaryType, mixins, and their supertypes
         concatRdfTypes();
+        if (resource() instanceof NonRdfSource) {
+            // gather versionability info from the parent
+            if (resource().getNode().getParent().isNodeType("mix:versionable")) {
```

Comment:
Fair enough. 
My question becomes, are we loosing versionability of NonRdfSource properties by storing them on the NonRdfSource? In other words, why is the check for "mix:versionable" not being performed on the NonRdfSource?: `resource().getNode().isNodeType(...)`